x-pack/test/api_integration/deployment_agnostic/apis/management/ingest_pipelines.ts

```diff
@@ -6,29 +6,34 @@
  */
 
 import expect from '@kbn/expect';
-
 import { IngestPutPipelineRequest } from '@elastic/elasticsearch/lib/api/types';
+import { DeploymentAgnosticFtrProviderContext } from '../../ftr_provider_context';
+import { SupertestWithRoleScopeType } from '../../services';
 
-import { FtrProviderContext } from '../../../ftr_provider_context';
-
-export default function ({ getService }: FtrProviderContext) {
-  const supertest = getService('supertest');
+export default function ({ getService }: DeploymentAgnosticFtrProviderContext) {
   const ingestPipelines = getService('ingestPipelines');
+  const roleScopedSupertest = getService('roleScopedSupertest');
   const log = getService('log');
+  let supertestWithAdminScope: SupertestWithRoleScopeType;
+
+  describe('Ingest Pipelines', function () {
+    before(async () => {
+      supertestWithAdminScope = await roleScopedSupertest.getSupertestWithRoleScope('admin', {
```

Comment:
Does this have the same privileges as `supertest` from the original test? Just wanted to make sure we don't give additional privileges that are not needed.

Reply:
`supertest` is inited with `system_indexes_superuser` role, which in stateful grants absolutely "everything". In the past it hided some bugs, and generally it is not recommended to use that role for API calls testing, but create something scoped to the particular functionality.

You can see both role definitions here https://github.com/elastic/kibana/blob/main/packages/kbn-es/src/stateful_resources/roles.yml#L99-L130 , `admin` is a default Cloud role (Cloud Org admin) and should be able to perform all the possible operations with Kibana.

But if you think ingesting pipeline should be available with lower permissions (`editor`) , we should definitely change it.

Reply:
Thanks for clarifying! I think leaving it with `admin` role is okay given that the privileges in the original test were even higher. cc: @alisonelizabeth